content/docs/contribute/git-hooks-guide.md

```diff
@@ -0,0 +1,47 @@
+---
+title: "Contributor Git Hooks Guide"
+date: 2024-03-20T15:56:27+02:00
+draft: false
+---
+
+# Integrating Pre-commit into Our Development Workflow
+
+## Overview
+
+Git hooks are scripts that Git executes before or after events such as: pre-commit, pre-push. These hooks are a powerful part of the Git ecosystem, enabling custom actions based on the current repository state or the changes being committed. They are used for a variety of tasks such as syntax checking, testing, linting, and auto-formatting code before it is committed or pushed. More information on git hooks is found [here](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)
+
+### Why Git hooks
+
+Git pipelines often fail because of linting issues, which then requires another commit to fix it, leaving the history of commits just filled with "liniting fix".
+
+## Why `pre-commit`
+
+Git hooks are not shared across the repo, and it's only avaiable locally. This means every developer will need to implement their own hooks. Hence tool called `pre-commit`. The tool name `pre-commit` name is misleading as it implies it only provides pre-commit hooks. However it also provides other stages such as `pre-push`.
+
+Through this tool, developers define a configuration file that specifies what 
+More information on the tool can be found [here](https://pre-commit.com/)
+
+
+## Setting Up:
+
+1. First, ensure that you have pre-commit installed. It can be installed via pip or Homebrew (for macOS users):
+
+```sh
+pip install pre-commit
+# or
+brew install pre-commit
+```
+
+2. **Install the git hooks scripts**:
+In Oakestra repo, there is a file called `.pre-commit-config.yaml` containing git hooks. They can be installed by running:
```

Comment:
I cannot find this file ?
![image](https://github.com/oakestra/documentation/assets/65814168/4cbcc6c1-37e2-4b54-b807-9a12fb848126)

Reply:
https://github.com/oakestra/oakestra/pull/279

PR is not yet merged.

Reply:
Waiting for this to be approved first before the merging the implementation PR.